plutus-core/plutus-core/src/PlutusCore/TypeCheck/Internal.hs

```diff
@@ -576,14 +576,12 @@ inferTypeM (Case ann resTy scrut branches) = do
             -- scrutinee does not have a SOP type with the right number of alternatives
             -- for the number of branches
             Nothing -> throwError (TypeMismatch ann (void scrut) expectedSop vScrutTy)
-        TyBuiltin _ someUni -> case annotateCaseBuiltin someUni branches of
+        ty -> case annotateCaseBuiltin ty branches of
```

Comment:
Please rename it to `vTy` to indicate that it's normalized (you can see all these comments about normalization, it's a consequence of us originally having trouble ensuring that normalization happens in the right places).

And update the PIR type checker too.